spec/features/user_logs_in_spec.rb

```diff
@@ -25,33 +25,57 @@
       visit '/login'
       fill_in('session[email]', with: 'example@example.com')
       click_link_or_button 'Login'
-      within("#errors") do
+      within(".flash") do
         expect(page).to have_content("Invalid Login")
       end
     end
 
-  end
+    scenario 'logging in does not clear the cart' do
+      user = User.create(full_name: 'dg',
```

Comment:
Use 2 (not 0) spaces for indentation.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.